### ISSUE TITLE
Thesis hyperparameters come from train.yaml, not the run's effective config — CLI overrides are invisible

### ISSUE THREAD

**Issue by kwojdalski** [kwojdalski/masters_thesis#816]
## Problem

`hyperparams.json` in every exported snapshot is built by `_load_scenario_hyperparams()` in `scripts/export_eval_to_thesis.py`, which reads `src/configs/scenarios/<scenario>/train.yaml` **directly**. A launch-time override (`--config-override training.gamma=0.95`) leaves no trace in that file, so the exported snapshot — and therefore the thesis — reports the *configured default*, not what the run actually used.

Concretely: train at `--config-override training.gamma=0.95` and the thesis still renders `γ = 0.9`.

## Scope of the gap

`max_steps` alone is already corrected (#797): the exporter reads the real checkpoint ladder via `_trained_steps_from_checkpoints()`, overwrites the YAML figure, keeps the original in `max_steps_configured`, and warns when they differ. **Every other hyperparameter has no such correction** — gamma, tau, learning rates, weight decay, policy noise, noise clip, policy delay, exploration noise, trading fees.

This now matters more than before: #813 and #815 wired ~31 prose figures to generated LaTeX macros, and the 13 hyperparameter macros read `hyperparams.json`. They are live with respect to the YAML but **not** with respect to the run, which is not what they claim to be.

## Evidence this is not hypothetical

Seven MLflow runs exist for `pooled_td3_hft_lob_state_space_pooled_streaming_selected_dsr`, and they disagree on `actor_weight_decay`:

```
run        actor_weight_decay  max_steps   gamma
74585ea1                  0.0    3000000     0.9
df7c20cb                  0.0    3000000     0.9
44b9726a                  0.0    3000000     0.9
d7c20092                2e-06    3000000     0.9
a74c112d                2e-06       1000     0.9
2b02c6ae                2e-06     200000     0.9
a9eea668                2e-06     600000     0.9
```

Four full-budget (3M-step) runs exist with **both** `0.0` and `2e-06`. The thesis currently states `λ = 2 × 10^{-6}` (from the YAML), but the snapshot records no `run_id` or `artifact_uri` — both are `null` in `latest_finished/run.json` — so **there is no way to determine which run produced the reported results**, and therefore no way to confirm the stated weight decay is the one used.

## The data already exists

`log_config_artifact()` in `src/trading_rl/callbacks/artifacts_config.py` already serialises the **resolved, post-override** config to MLflow. Its own docstring: *"Always serializes the in-memory config (including any CLI overrides) to `config/effective_config.yaml`"*. Confirmed on disk, e.g.:

```
mlruns/21/74585ea15f4844dcb4dbc61f5fde1839/artifacts/config/effective_config_78hvrs8d.yaml
```

(The filename carries a random suffix because `mlflow.log_artifact()` is handed a `NamedTemporaryFile` name, so it is glob-discoverable rather than fixed.)

The exporter simply never reads it. `params.json` in the snapshot is written as an empty `{}` placeholder.

## Proposed fix

1. In `export_eval_to_thesis.py`, resolve the run's MLflow artifact directory and copy `config/effective_config_*.yaml` into the snapshot as `effective_config.yaml`.
2. Build `hyperparams.json` from that effective config, falling back to `train.yaml` only when it is unavailable, and record which source was used.
3. Record `run_id` and `artifact_uri` in `run.json` so a snapshot can be traced to the run that produced it.
4. Keep the existing `max_steps` checkpoint-ladder correction — a run stopped early still needs the outcome, not the intent.

This makes the thesis report what was actually run, and gives every wired macro a verifiable provenance chain.
